### DISCUSSION TITLE
Ideas for a Bob UI

### DISCUSSION THREAD

**Discussion by lispyclouds** [bob-cd/bob#110] [Ideas]
The rough ideas I was having for a Bob UI are:
- Be in the same spirit of being made of re-usable components as Bob
- [Nice to have] Have an ability to export UI modules/components such that users are able to "Build their own UI"
- Can it somehow be a [WYSIWYG](https://en.wikipedia.org/wiki/WYSIWYG) editor too? Spirit of the REPL!
- Happy to have it in any language!

Tracking issue: https://github.com/bob-cd/bob/issues/40

**Comment by lispyclouds** [comment]
Having said that, lets start something small and simple, like being able to show the current state of things, then move onto to edit things and beyond!

**Reply by lispyclouds** [reply]
Also something along the lines of [Kibana](https://www.elastic.co/kibana/) could be a good way of visualising and controlling the bob cluster, I think!

**Reply by TimoKramer** [reply]
I would love to see something like [this ](https://github.com/prestancedesign/babashka-htmx-todoapp) running on babashka. Appears to be low-tech and bob-like. Don't know how to make re-usable components though.

As I understand it, this would be a counterpart to wendy, right? So it would be cool to have it data-driven and auto-generated (don't know if this is possible).

**Reply by imimlatu** [reply]
HTMX uses regular HTML as the engine of application state, so [Web Components](https://developer.mozilla.org/en-US/docs/Web/Web_Components) along with a server with a basic REST API serving HTMX-powered content would be very simple and straightforward. I immediately thought "HTMX fits the bob style" when I saw this issue, and I wasn't disappointed when reading where discussion left off from last year!

Is this something that there is still interest in?

**Reply by TimoKramer** [reply]
Hey,
absolutely. I was thinking about how to start with it. It is not a small endeavor for a non-UI person like me. We should probably cut it down into small pieces...
- Reading the pipelines and showing them sorted by last run
- Detailed view of pipelines with logs
- ...

**Comment by TimoKramer** [comment]
I've built the long awaited Bob UI, at least partially. It's maybe 80% complete: https://github.com/bob-cd/muck
Used Java because I wanted to write some modern Java just to see how it compares to what I've learnt in Uni.
